airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -215,8 +220,8 @@ private void replicate(final Path jobRoot,
             }
           });
 
-      final CompletableFuture<?> readSrcAndWriteDstThread = CompletableFuture.runAsync(
-          readFromSrcAndWriteToDstRunnable(
+      final CompletableFuture readSrcAndWriteDstThread =
+          srcHeartbeatTimeoutChaperone.runWithHeartbeatThread(CompletableFuture.runAsync(readFromSrcAndWriteToDstRunnable(
```

Comment:
Is this the part that differed from what we discussed? If so, I am curious to understand why we still need to wrap the current future.